<?xml version="1.0"?>
<launch> 
  <arg name="rviz" default="false"/>
  <arg name="gui" default="false" />
  <arg name="debug_image" default="true" />
  <arg name="camera_info" default="/camera_info" />
  <arg name="debug_image_topic" default="debug_image" />
  <arg name="image_topic" default="/camera/image" />
  <arg name="image_transport" default="raw" />
  <arg name="marker_size" default="0.1095" />
  <arg name="calibration_file" default="$(find aruco_mapping)/data/raspicam_v1_underwater.ini" />

  <!-- RVIZ -->
  <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(find aruco_mapping)/launch/aruco_config.rviz" />

  <!-- ArUco mapping -->
  <node pkg="aruco_mapping" type="aruco_mapping_fixed" name="aruco_mapping" output="screen">
    <param name="image_topic" type="string" value="$(arg image_topic)"/>
    <param name="image_transport" type="string" value="$(arg image_transport)" />
    <param name="calibration_file" type="string" value="$(arg calibration_file)"/>
    <param name="marker_size" type="double" value="$(arg marker_size)"/>
    <param name="gui" type="bool" value="$(arg gui)" />
    <param name="camera_info" type="string" value="$(arg camera_info)" />
    <param name="debug_image" type="bool" value="$(arg debug_image)" />
    <param name="debug_image_topic" type="string" value="$(arg debug_image_topic)" />

  </node>  
</launch>
